docs/developer-guide/architecture.md

```diff
@@ -0,0 +1,78 @@
+# Architecture
+
+HAFO follows a simple, focused architecture designed for extensibility.
+
+## Overview
+
+```
```

Comment:
Trying to make a mermaid diagram here? Also don't enumerate the forecasters in the main file.